asserts/model.go

```diff
@@ -268,46 +568,80 @@ func assembleModel(assert assertionBase) (Assertion, error) {
 		}
 	}
 
-	// kernel/gadget must be valid snap names and can have (optional) tracks
-	// - validate those
-	if err := checkSnapWithTrackHeader("kernel", assert.headers); err != nil {
-		return nil, err
-	}
-	if err := checkSnapWithTrackHeader("gadget", assert.headers); err != nil {
-		return nil, err
+	if !extended {
```

Comment:
Empty if statement? This looks fishy.

Reply:
thanks for spotting this, it's a left over from wip code, all the relevant code is now in the `if extented {...} else {...}` below